layer/trace_layer.cpp

```diff
@@ -53,6 +53,29 @@ const VkLayerProperties kLayerProps = {
 
 const std::vector<VkExtensionProperties> kDeviceExtensionProps = { VkExtensionProperties{ "VK_EXT_tooling_info", 1 } };
 
+/// An alphabetical list of device extensions which we do not report upstream if
+/// other layers or ICDs expose them to us.
+const char* const kUnsupportedDeviceExtensions[] = {
+    // Supporting the CPU moving around descriptor set data directly has too many
+    // perf/robustness tradeoffs to be worth it.
+    VK_VALVE_DESCRIPTOR_SET_HOST_MAPPING_EXTENSION_NAME,
+};
+
+static void remove_extensions(std::vector<VkExtensionProperties>& extensionProps,
+                              const char* const                   screenedExtensions[],
+                              const size_t                        screenedCount)
+{
+    auto new_end = std::remove_if(
+        extensionProps.begin(),
+        extensionProps.end(),
+        [&screenedExtensions, screenedCount](const VkExtensionProperties& extension) {
+            return std::find_if(screenedExtensions, &screenedExtensions[screenedCount], [&extension](auto screened) {
```

Comment:
As a future consideration I personally have trouble reading nested lambdas, but I think this is fine for getting this in and closing the issue